lib/varmap.js

```diff
@@ -0,0 +1,69 @@
+'use strict'
+var util = require('./util')
+
+function VarMap (lengthType, keyType, valueType) {
+  if (!util.isAbstractCodec(lengthType)) throw new TypeError('lengthType is invalid codec')
+  if (!util.isAbstractCodec(keyType)) throw new TypeError('keyType is invalid codec')
+  if (!util.isAbstractCodec(valueType)) throw new TypeError('valueType is invalid codec')
+
+  function _length (object) {
+    if (!object) throw new TypeError('Expected object')
+
+    var size = 0
+    var i = 0
+    for (var key in object) {
```

Comment:
@fanatid I suppose I figure I've never personally found a situation where I pass an object that has a `prototype` to this sort of function;  but I also avoid complex classes wherever possible.